plugins/redfish/fu-redfish-device.c

```diff
@@ -512,19 +512,106 @@ fu_redfish_device_get_backend (FuRedfishDevice *self)
+	    g_strcmp0 (message_id, "TaskEvent.1.0.TaskCompletedWarning") == 0 ||
+	    g_strcmp0 (message_id, "TaskEvent.1.0.TaskCompletedOK") == 0 ||
+	    g_strcmp0 (message_id, "Base.1.6.Success") == 0)
+		return TRUE;
+
+	/* set flags */
+	if (g_strcmp0 (message_id, "Base.1.10.ResetRequired") == 0) {
+		fu_device_add_flag (FU_DEVICE (self), FWUPD_DEVICE_FLAG_NEEDS_REBOOT);
+		return TRUE;
+	}
+
+	/* set error code */
+	if (g_strcmp0 (message_id, "Update.1.0.AwaitToActivate") == 0) {
+		ctx->error_code = FWUPD_ERROR_NEEDS_USER_ACTION;
+		return TRUE;
+	}
+	if (g_strcmp0 (message_id, "Update.1.0.TransferFailed") == 0) {
+		ctx->error_code = FWUPD_ERROR_WRITE;
+		return TRUE;
+	}
+	if (g_strcmp0 (message_id, "Update.1.0.ActivateFailed") == 0) {
+		ctx->error_code = FWUPD_ERROR_INVALID_FILE;
+		return TRUE;
+	}
+	if (g_strcmp0 (message_id, "Update.1.0.VerificationFailed") == 0 ||
+	    g_strcmp0 (message_id, "LenovoFirmwareUpdateRegistry.1.0.UpdateVerifyFailed") == 0) {
+		ctx->error_code = FWUPD_ERROR_INVALID_FILE;
+		return TRUE;
+	}
+	if (g_strcmp0 (message_id, "Update.1.0.ApplyFailed") == 0) {
+		ctx->error_code = FWUPD_ERROR_WRITE;
+		return TRUE;
+	}
+
+	/* set status */
+	if (g_strcmp0 (message_id, "Update.1.1.TargetDetermined") == 0) {
+		fu_device_set_status (FU_DEVICE (self), FWUPD_STATUS_LOADING);
+		return TRUE;
+	}
+	if (g_strcmp0 (message_id, "LenovoFirmwareUpdateRegistry.1.0.UpdateAssignment") == 0) {
+		fu_device_set_status (FU_DEVICE (self), FWUPD_STATUS_LOADING);
+		return TRUE;
+	}
+	if (g_strcmp0 (message_id, "LenovoFirmwareUpdateRegistry.1.0.PayloadApplyInProgress") == 0) {
+		fu_device_set_status (FU_DEVICE (self), FWUPD_STATUS_DEVICE_WRITE);
+		return TRUE;
+	}
+	if (g_strcmp0 (message_id, "LenovoFirmwareUpdateRegistry.1.0.PayloadApplyCompleted") == 0) {
+		fu_device_set_status (FU_DEVICE (self), FWUPD_STATUS_IDLE);
+		return TRUE;
+	}
+	if (g_strcmp0 (message_id, "LenovoFirmwareUpdateRegistry.1.0.UpdateVerifyInProgress") == 0) {
+		fu_device_set_status (FU_DEVICE (self), FWUPD_STATUS_DEVICE_VERIFY);
+		return TRUE;
+	}
+	if (g_strcmp0 (message_id, "Update.1.1.TransferringToComponent") == 0) {
+		fu_device_set_status (FU_DEVICE (self), FWUPD_STATUS_LOADING);
+		return TRUE;
+	}
+	if (g_strcmp0 (message_id, "Update.1.1.VerifyingAtComponent") == 0) {
+		fu_device_set_status (FU_DEVICE (self), FWUPD_STATUS_DEVICE_VERIFY);
+		return TRUE;
+	}
+	if (g_strcmp0 (message_id, "Update.1.1.UpdateInProgress") == 0) {
+		fu_device_set_status (FU_DEVICE (self), FWUPD_STATUS_DEVICE_WRITE);
+		return TRUE;
+	}
+	if (g_strcmp0 (message_id, "Update.1.1.UpdateSuccessful") == 0) {
+		fu_device_set_status (FU_DEVICE (self), FWUPD_STATUS_IDLE);
+		return TRUE;
+	}
+	if (g_strcmp0 (message_id, "Update.1.1.InstallingOnComponent") == 0) {
+		fu_device_set_status (FU_DEVICE (self), FWUPD_STATUS_DEVICE_WRITE);
+		return TRUE;
+	}
+	g_warning ("no idea how to process MessageId %s", message_id);
+	return TRUE;
```

Comment:
shouldn't this be FALSE and an error?

Reply:
it's not critical if we don't know what to do with the message, it could be something like HpeFirmwareUpdateRegistry.1.0.CheckLicenseExpiration or something. Maybe it's best if that function just doens't return anything and we drop the GError?

Reply:
seems fine to me

Reply:
Okay, fixed.